src/test/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJModuleTest.java

```diff
@@ -0,0 +1,61 @@
+package fi.aalto.cs.apluscourses.intellij.model;
+
+import static java.lang.Thread.sleep;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.io.FileUtilRt;
+import com.intellij.testFramework.HeavyPlatformTestCase;
+import fi.aalto.cs.apluscourses.TestHelper;
+import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
+import fi.aalto.cs.apluscourses.model.Module;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.time.Instant;
+import org.junit.Test;
+
+public class IntelliJModuleTest extends HeavyPlatformTestCase implements TestHelper {
+
+  @Test
+  public void testHasLocalChangesReturnsTrue() throws IOException, InterruptedException {
+    //  given
+    Project project = getProject();
+
+    IntelliJModelFactory factory = new IntelliJModelFactory(project);
+    Module module = factory.createModule("first", new URL("http://firstURL"), "1");
+    final IntelliJModule intelliJModule = (IntelliJModule) module;
+
+    File tempDirectory = FileUtilRt.createTempDirectory("first", "", true);
+    File tempFileOne = FileUtilRt.createTempFile(tempDirectory, "test1", ".json", true);
+    File tempFileTwo = FileUtilRt.createTempFile(tempDirectory, "test2", ".json", true);
+    tempFileTwo.setLastModified(Instant.now().toEpochMilli());
+    long oneSecond = 1000L;
+    sleep(PluginSettings.REASONABLE_DELAY_FOR_MODULE_INSTALLATION + oneSecond);
+
+    long now = Instant.now().toEpochMilli();
+    tempFileOne.setLastModified(now);
```

Comment:
Sonar sees vulnerabilities where `setLastModified` is called while ignoring the return value. As a solution to this, I suggest surrounding each of these with `assertTrue(...)`. Should it happen that the file metadata cannot be updated, the test results would become meaningless, so it is reasonable to fail anyways. Well, it's not probable that would ever happen but at least Sonar would be happy.